File: MapAYakApi/Models/Dtos/Data/LocationDto.cs
using MapAYakApi.Models;

namespace MapAYakApi.Dtos.Data
{
    public class LocationDto
    {
        public string UserName { get; set; }

        public LocationType Type { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public decimal Latitude { get; set; }

        public decimal Longitude { get; set; }
    }
}

File: MapAYak/src/app/core/dtos/map/location.dto.ts
import { LocationType } from "../../enums/core.enums";

export class LocationDto {
    public userName: string;
    public type: LocationType;
    public name: string;
    public description: string;
    public latitude: number;
    public longitude: number;

    public constructor(initial?: Partial<LocationDto>) {
        Object.assign(this, initial);
    }
}
